core/classes/Core/User.php

```diff
@@ -472,7 +477,16 @@ public function getSignature(): string {
      * @return string URL to their avatar image.
      */
     public function getAvatar(int $size = 128, bool $full = false): string {
-        return AvatarSource::getAvatarFromUserData($this->data(), $this->hasPermission('usercp.gif_avatar'), $size, $full);
+        $data = $this->data();
+
+        $integrationUser = $this->getIntegration('Minecraft');
+        if ($integrationUser != null) {
+            $data->uuid = $integrationUser->data()->identifier;
+        } else {
+            $data->uuid = '';
+        }
+
+        return AvatarSource::getAvatarFromUserData($data, $this->hasPermission('usercp.gif_avatar'), $size, $full);
```

Comment:
hmm, i wonder how this will impact the use of the `getAvatarFromUserData` method when we use it without the entire user object...